pkg/sql/sem/builtins/builtins.go

```diff
@@ -2160,6 +2160,9 @@ may increase either contention or retry errors, or both.`,
 			_, err := tree.ExactCtx.Floor(&dd.Decimal, x)
 			return dd, err
 		}, "Calculates the largest integer not greater than `val`."),
+		integerOverload1(func(x int64) (tree.Datum, error) {
+			return tree.NewDInt(tree.DInt(x)), nil
+		}, "Calculates the largest integer not greater than `val`."),
```

Comment:
the return type of this function is float --
```
otan=# select pg_typeof(floor(2::int));
    pg_typeof
------------------
 double precision
(1 row)
```

also, do you mind making this work for floating point arithmetic?

postgres:
```
otan=# select pg_typeof(floor(2::float));
    pg_typeof
------------------
 double precision
(1 row)
```

with your change i think it still doesn't work

same with ceil!

Reply:
Thanks for the review, and sorry about the mistake. I've tried to make a fix for this.

Not sure if i'm understanding the part about floating point arithmetic correctly. Running `select pg_typeof(floor(2::float));` returns `float` with the current implementation. Could you provide a test case to show what you mean?

Reply:
> sorry about the mistake
don't be sorry :) very hard to spot!

> Not sure if i'm understanding the part about floating point arithmetic correctly. Running select pg_typeof(floor(2::float)); returns float with the current implementation. Could you provide a test case to show what you mean?

i noticed you added a test for it and it was already correct (i missed that part of the code), so disregard that comment.

thanks again for doing this.